python/src/dynamixel_sdk/protocol1_packet_handler.py

```diff
@@ -325,162 +322,154 @@ def readRx(self, port, dxl_id, length):
         data, result, error = self.readRx(port, dxl_id, 1)
         data_read = data[0] if (result == COMM_SUCCESS) else 0
-        return data_read, result, error  
+        return data_read, result, error
+
     def read1ByteTxRx(self, port, dxl_id, address):
         data, result, error = self.readTxRx(port, dxl_id, address, 1)
         data_read = data[0] if (result == COMM_SUCCESS) else 0
         return data_read, result, error
 
     def read2ByteTx(self, port, dxl_id, address):
         return self.readTx(port, dxl_id, address, 2)
+
     def read2ByteRx(self, port, dxl_id):
         data, result, error = self.readRx(port, dxl_id, 2)
         data_read = DXL_MAKEWORD(data[0], data[1]) if (result == COMM_SUCCESS) else 0
         return data_read, result, error
+
     def read2ByteTxRx(self, port, dxl_id, address):
         data, result, error = self.readTxRx(port, dxl_id, address, 2)
         data_read = DXL_MAKEWORD(data[0], data[1]) if (result == COMM_SUCCESS) else 0
         return data_read, result, error
 
     def read4ByteTx(self, port, dxl_id, address):
         return self.readTx(port, dxl_id, address, 4)
+
     def read4ByteRx(self, port, dxl_id):
         data, result, error = self.readRx(port, dxl_id, 4)
         data_read = DXL_MAKEDWORD(DXL_MAKEWORD(data[0], data[1]),
                                   DXL_MAKEWORD(data[2], data[3])) if (result == COMM_SUCCESS) else 0
         return data_read, result, error
+
     def read4ByteTxRx(self, port, dxl_id, address):
         data, result, error = self.readTxRx(port, dxl_id, address, 4)
         data_read = DXL_MAKEDWORD(DXL_MAKEWORD(data[0], data[1]),
                                   DXL_MAKEWORD(data[2], data[3])) if (result == COMM_SUCCESS) else 0
         return data_read, result, error
 
     def writeTxOnly(self, port, dxl_id, address, length, data):
-        result = COMM_TX_FAIL
-
         txpacket = [0] * (length + 7)
 
         txpacket[PKT_ID] = dxl_id
         txpacket[PKT_LENGTH] = length + 3
         txpacket[PKT_INSTRUCTION] = INST_WRITE
         txpacket[PKT_PARAMETER0] = address
 
-        txpacket[PKT_PARAMETER0 + 1 : PKT_PARAMETER0 + 1 + length] = data[0 : length]
+        txpacket[PKT_PARAMETER0 + 1: PKT_PARAMETER0 + 1 + length] = data[0: length]
 
         result = self.txPacket(port, txpacket)
         port.is_using = False
 
         return result
 
     def writeTxRx(self, port, dxl_id, address, length, data):
-        result = COMM_TX_FAIL
-
         txpacket = [0] * (length + 7)
-        rxpacket = None
 
         txpacket[PKT_ID] = dxl_id
         txpacket[PKT_LENGTH] = length + 3
         txpacket[PKT_INSTRUCTION] = INST_WRITE
         txpacket[PKT_PARAMETER0] = address
 
-        txpacket[PKT_PARAMETER0 + 1 : PKT_PARAMETER0 + 1 + length] = data[0 : length]
+        txpacket[PKT_PARAMETER0 + 1: PKT_PARAMETER0 + 1 + length] = data[0: length]
         rxpacket, result, error = self.txRxPacket(port, txpacket)
 
-        del rxpacket[:]; del rxpacket
```

Comment:
Not needed at all. Python garbage collector takes care of this

Reply:
@roncapat 

That's very welcomed.

Reply:
@roncapat 

but one thing that I concerned is that,
the garbage collector works every time when the function is stopped running?
or only when the script is ended?

Reply:
The garbage collector cleans memory as soon as an object reference count reaches 0. You're definetly not going to get a memory leak by not explicitly freeing memory, python takes care of it 99.9% of times.

Reply:
@roncapat 

how about micropython?
it would be the best if this can be compatible to the arduino level

Reply:
Micropython has a VM, it does not run directly on hardware, so the runtime takes care of GC too. Maybe you can consider gc.collect() calls for triggering periodic cleanup, but explicitly deleting before exiting a function, and thus loosing the only reference of those objects is not useful because GC should be able to clean that situation.